### ISSUE TITLE
Windows 11, Visual Studio build, QGIS.exe built, but will not run

### ISSUE THREAD

**Issue by danodom-tcb** [qgis/QGIS#59249]
### What is the bug or the crash?

I have successfully built QGIS.exe with Visual Studio with some great assistance from @m-kuhn (thank you).  However, I cannot get the .exe to run in Windows from the bin folder.   QGIS goes to startup, and I can see something like "Starting Python" and then the window just goes away..

Also, I can take the built .exe and put it in the "QGIS Install Folder from the Installer", create an .env file in the bin folder, and it works.  So, the executable is good, but there appears to be a few things missing.   I am betting it needs the .env file, which I started, but there appears to be gaps in what's in the install folder vs what's in the SDK or some things might be the same, but referenced by a different name. . 

I am running Windows 11, with Visual Studio 2022, SDK(vcpkg-export-20241015-154141).

### Steps to reproduce the issue

see above.  

### Versions

Close to latest version of source, post install.md update referencing the SDK

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by nicogodet** [comment]
I looked at your other issues and, in my opinion, your build environment is kind a mess 😄 
And it makes it very difficult to debug...

You should proceed step by step. If you encounter an error, report it (make sure this is a real issue, double check your cmake command, etc...)

**Comment by danodom-tcb** [comment]
> I looked at your other issues and, in my opinion, your build environment is kind a mess 😄 And it makes it very difficult to debug...
> 
> You should proceed step by step. If you encounter an error, report it (make sure this is a real issue, double check your cmake command, etc...)

@nicogodet It was initially a clean system following the build instructions but I've had to add things that were missing to get past build issues as there are gaps in the documentation.    Literally, this was a completely clean system before I started and I started with the install.md instructions but those did not work.  I finally have a build and @m-kuhn  has been helping get through some issues.

**Comment by nicogodet** [comment]
> It was initially a clean system following the build instructions but I've had to add things that were missing to get past build issues

This is wrong. I try to build locally too and I assumed that Install.md is perfect and everything is correct. I just followed the steps and reported errors when I found some : 

- https://github.com/qgis/QGIS/issues/59231
- https://github.com/qgis/QGIS/issues/59211
- https://github.com/open-vcpkg/python-registry/issues/60
- https://github.com/qgis/QGIS/issues/59157#issuecomment-2437998533
- https://github.com/qgis/QGIS/issues/59157#issuecomment-2438247820

I believe this is the correct way to proceed instead of locally install missing things...

**Comment by nicogodet** [comment]
Currently, with fixes, I can follow Install.md steps without errors when building dependencies locally (I expect bigobj to raise an error soon)

**Comment by nicogodet** [comment]
Successful build fix bigobj workaround (waiting for automoc)
Need to figure out .env file now...

**Comment by danodom-tcb** [comment]
>>I believe this is the correct way to proceed instead of locally install missing things...
I agree with you here and have a thread open working with m-kuhn as we go along.  Please keep in mind that I am primarily a windows guy, and not really familiar with python, but coming up to speed.. So, there was a lot of things I was learning and picking up on the fly.

I did follow the instructions, and even added missing parameters to the cmake for the missing flex and bison parameters that in the command, so intentions were good, I just didn't know what I didn't know.

So, after the build worked, do you think it's basically the missing .env file causing most of the issues with it not being able to run?  That's my thought, but I am not 100% sure I can align everything in the installed environment file vs. what's in the sdk.  That, I'll need some help with and some guidance.

Also, my plan is going to go back from square one and see if I can get a repeatable build and run process, that's the goal.

Please let me know about what you find with getting the built .exe to run.


________________________________

**Comment by danodom-tcb** [comment]
Also, could provide me with what all needs to be installed, currently the instructions say Visual Studio 2022 and then the SDK and basically that's it.

What else needs to be installed as I am going to be working on getting a clean system to use for starting this from scratch again?

From the outside looking in, the install.md can be a tad confusing on what's required based on which method you are using.
________________________________

**Comment by nicogodet** [comment]
(side note: Best would be to answer on github directly instead of email which tends to break message format)

>Also, could provide me with what all needs to be installed, currently the instructions say Visual Studio 2022 and then the SDK and basically that's it.
>
>What else needs to be installed as I am going to be working on getting a clean system to use for starting this from scratch again?

You only need visual studio and you need to compile dependencies yourself as SDK is broken (see https://github.com/qgis/QGIS/issues/59231).

Cmake command looks like:

```ps
cmake -S . `
	  -B build `
	  -D WITH_VCPKG=ON `
	  -D BUILD_WITH_QT6=ON `
	  -D WITH_QTWEBKIT=OFF `
	  -D VCPKG_TARGET_TRIPLET=x64-windows-release `
	  -D VCPKG_HOST_TRIPLET=x64-windows-release `
	  -D CMAKE_BUILD_TYPE=RelWithDebInfo `
	  -D WITH_DESKTOP=ON `
	  -D WITH_3D=ON `
	  -D WITH_BINDINGS=ON `
	  -D ENABLE_TESTS=OFF `
	  -D USE_CCACHE=ON `
	  -D CMAKE_C_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
	  -D CMAKE_CXX_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
	  -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:\vcpkg\src" `
	  -D FLEX_EXECUTABLE="C:\Flex_bison\win_flex.exe" `
	  -D BISON_EXECUTABLE="C:\Flex_bison\win_bison.exe" `
	  -D CREATE_ZIP=ON
```

**Comment by nicogodet** [comment]
@m-kuhn I'm able to confirm a successful build on windows with below steps:

1. Install VS2022
2. ```ps
    cmake -S . `
          -B build `
          -G Ninja `
          -D WITH_VCPKG=ON `
          -D BUILD_WITH_QT6=ON `
          -D WITH_QTWEBKIT=OFF `
          -D VCPKG_TARGET_TRIPLET=x64-windows-release `
          -D VCPKG_HOST_TRIPLET=x64-windows-release `
          -D CMAKE_BUILD_TYPE=RelWithDebInfo `
          -D WITH_DESKTOP=ON `
          -D WITH_3D=ON `
          -D WITH_BINDINGS=ON `
          -D ENABLE_TESTS=OFF `
          -D USE_CCACHE=ON `
          -D CMAKE_C_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
          -D CMAKE_CXX_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
          -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:\vcpkg\src" `
          -D FLEX_EXECUTABLE="C:\Flex_bison\win_flex.exe" `
          -D BISON_EXECUTABLE="C:\Flex_bison\win_bison.exe" `
          -D CREATE_ZIP=ON `
          -D CMAKE_INSTALL_PREFIX="D:\Qgis_local"
    ```
    - Use CREATE_ZIP=ON to simplify the qgis execution later (my own choice)
    - Specify CMAKE_INSTALL_PREFIX to avoid error later
3. ```ps
    cmake --build build --config RelWithDebInfo --parallel
    ```
4. ```ps
    cmake --build build --config RelWithDebInfo --target bundle
    ```
5. Execute `.\build\_CPack_Packages\win64\ZIP\qgis-3.39.0-win64\bin\qgis.exe`

**Comment by m-kuhn** [comment]
:tada: 

> Specify CMAKE_INSTALL_PREFIX to avoid error later

where did that come from?

**Comment by nicogodet** [comment]
https://github.com/qgis/QGIS/issues/59267

CMake tries to install icons in default install dir (C:/Program Files (x86)) without admin rights

**Comment by danodom-tcb** [comment]
@nicogodet  @m-kuhn , so I don't need to install python or anything else, only Visual Studio 2022?

**Comment by danodom-tcb** [comment]
@nicogodet @m-kuhn and do not build from within VS 2022 (debug or release)?

**Comment by m-kuhn** [comment]
> @nicogodet @m-kuhn , so I don't need to install python or anything else, only Visual Studio 2022?

and flex and bison

> @nicogodet @m-kuhn and do not build from within VS 2022 (debug or release)?

I would avoid doing so in a first step (just to avoid problems); but it should also work

**Comment by danodom-tcb** [comment]
> and flex and bison

Assume the link to winflexbison in the install.md is appropriate for these?

**Comment by nicogodet** [comment]
> Assume the link to winflexbison in the install.md is appropriate for these?

Yes

**Comment by danodom-tcb** [comment]
@nicogodet  is this the root path to the QGIS source code?

-D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=**C:\vcpkg\src**" `

**Comment by nicogodet** [comment]
No.

My QGIS src are in D:\QGIS_src and all commands from https://github.com/qgis/QGIS/issues/59249#issuecomment-2446511634 are executed within this folder.

-D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:\vcpkg\src" allows you to specify where vcpkg will store downloaded dependencies. I think you can safely omit this option.

**Comment by m-kuhn** [comment]
This is there to avoid "long paths". Windows has a limitation (of 260?) chars and with some build paths this can cause problems. Or spaces in the path. You can try without, but it may cause issues

**Comment by nicogodet** [comment]
Indeed. Even with long path modification in regedit I recall build failure because of too long path.

**Comment by danodom-tcb** [comment]
> No.
> 
> My QGIS src are in D:\QGIS_src and all commands from [#59249 (comment)](https://github.com/qgis/QGIS/issues/59249#issuecomment-2446511634) are executed within this folder.
> 
> -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:\vcpkg\src" allows you to specify where vcpkg will store sources of downloaded dependencies. I think you can safely omit this option.

@m-kuhn @nicogodet so, do I need to use this or not?  if so, it looks like I need to also download the current SDK.

fyi, I have a net new server and starting the process of getting vs and other things installed to start/test the build/execute process.

**Comment by danodom-tcb** [comment]
> > you should set it, it needs to be a short path (not sure exactly, fewer than 10 chars or so) which is writable for you. at the moment you should avoid the SDK it is not fully working. you do not need to download it.
> 
> @m-kuhn . but isn't this the path to the SDK? or maybe I am misinterpreting
> 
> -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=**C:\vcpkg\src**"

Is this a local that will be created and written to, I can add it in but want to make sure I provision correctly, if any is needed.

**Comment by danodom-tcb** [comment]
@m-kuhn @nicogodet 

Is this something I need to download or an option for the visual studio install (but I didn't see it)?

       -D CMAKE_C_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
       -D CMAKE_CXX_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `

I don't have a ccache folder and the exe is not anywhere in the VS installed location tree..

**Comment by nicogodet** [comment]
> but isn't this the path to the SDK? or maybe I am misinterpreting
> 
> -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=**C:\vcpkg\src**"

This should point to an empty directory where your user can create folders and files

> Is this something I need to download or an option for the visual studio install (but I didn't see it)?
> 
> ```
>    -D CMAKE_C_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
>    -D CMAKE_CXX_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
> ```
> 
> I don't have a ccache folder and the exe is not anywhere in the VS installed location tree..

You can remove this option.
See https://ccache.dev/ for more info (this is a tool you need to manually install)

**Comment by danodom-tcb** [comment]
> > I don't have a ccache folder and the exe is not anywhere in the VS installed location tree..
> 
> You can remove this option. See https://ccache.dev/ for more info (this is a tool you need to manually install)

@nicogodet , I found it and went ahead and downloaded it, trying to follow as close to exact as I can ..

I've started the first cmake to build everything out...  wish me luck  :)

**Comment by danodom-tcb** [comment]
@nicogodet @m-kuhn I received errors on the first cmake, I'll post the manifest.

**Comment by danodom-tcb** [comment]
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: C:/QGIS/QGIS/build/vcpkg_installed/x64-windows-release/tools/python3/python.exe -m installer --prefix C:/QGIS/QGIS/build/vcpkg_installed/x64-windows-release/tools/python3 --destdir C:/QGIS/QGIS/build/vcpkg_installed/vcpkg/pkgs/py-setuptools_x64-windows-release C:/QGIS/QGIS/build/vcpkg_installed/vcpkg/pkgs/py-setuptools_x64-windows-release/wheels/setuptools-72.1.0-py3-none-any.whl
    Working Directory: C:/QGIX/vcpkg/src/py-setuptools
    Error code: 1
    See logs for more information:
      C:\QGIX\vcpkg\src\py-setuptools\python-installer-x64-windows-release-err.log

**Comment by danodom-tcb** [comment]
Here is the 
[vcpkg-manifest-install.txt](https://gist.github.com/danodom-tcb/63c46d3b72b93b542d25dea925583c2b)

[python-installer-x64-windows-release-err.txt](https://gist.github.com/danodom-tcb/f6c9bc3a59915d0c75037a0403650c0d)

**Comment by nicogodet** [comment]
-D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:\QGIX\vcpkg\src"

C:\QGIX -> C:\QGIS ?

**Comment by nicogodet** [comment]
See https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later
And retry

**Comment by danodom-tcb** [comment]
> See https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later And retry

@nicogodet ,
ok, I'll look, but I would suggest adding the link to ccache download and this link to the install.md file as well.  :)

**Comment by nicogodet** [comment]
ccache is not mandatory 

Link for long path could be added indeed

**Comment by danodom-tcb** [comment]
> ccache is not mandatory
> 
> Link for long path could be added indeed

How is ccache.exe used when it's included?

**Comment by danodom-tcb** [comment]
@nicogodet @m-kuhn , I have to admit, this is the first time I've seen a reference to fortran.. 

![image](https://github.com/user-attachments/assets/7574a078-a3b8-48b0-b9bd-412110cf57c2)


Also, how long does this step take in the process as it's just been sitting here for like 10+ minutes.
![image](https://github.com/user-attachments/assets/6f78b7df-54b7-4318-bfc3-8161cde66c6a)

**Comment by m-kuhn** [comment]
prepare for hours

**Comment by nicogodet** [comment]
It took between 1 and 2 hours on an AMD Threadripper 32C/64T.

**Comment by danodom-tcb** [comment]
> It took between 1 and 2 hours on an AMD Threadripper 32C/64T.

first step completed in 169 minutes... no errors thankfully..

**Comment by danodom-tcb** [comment]
@m-kuhn  @nicogodet I got a ton of errors on the last step
>>cmake --build build --config RelWithDebInfo --target bundle

This looks like maybe some sort of path issue as it looks like it failed adding every file into the cpack.


tons of these : 
  CMake Error at C:/QGIS/QGIS/build/src/auth/basic/cmake_install.cmake:39 (file):
    file INSTALL cannot find
    "C:/QGIS/QGIS/build/output/plugins/Release/authmethod_basic.dll": File
    exists.
  Call Stack (most recent call first):
    C:/QGIS/QGIS/build/src/auth/cmake_install.cmake:37 (include)
    C:/QGIS/QGIS/build/src/cmake_install.cmake:38 (include)
    C:/QGIS/QGIS/build/cmake_install.cmake:285 (include)


  CMake Error at C:/QGIS/QGIS/build/src/auth/esritoken/cmake_install.cmake:39 (file):
    file INSTALL cannot find
    "C:/QGIS/QGIS/build/output/plugins/Release/authmethod_esritoken.dll": File
    exists.
  Call Stack (most recent call first):
    C:/QGIS/QGIS/build/src/auth/cmake_install.cmake:38 (include)
    C:/QGIS/QGIS/build/src/cmake_install.cmake:38 (include)
    C:/QGIS/QGIS/build/cmake_install.cmake:285 (include)

CUSTOMBUILD : CPack error : Error when generating package: qgis [C:\QGIS\QGIS\build\bundle.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): e
rror MSB8066: Custom build for 'C:\QGIS\QGIS\build\CMakeFiles\4f936f24e2671abb107f35661b43692c\bundle.rule;C:\QGIS\QGIS
\CMakeLists.txt' exited with code 1. [C:\QGIS\QGIS\build\bundle.vcxproj]

**Comment by m-kuhn** [comment]
It seems it picks up `Release` instead of `RelWithDebInfo` when installing the targets from `src/auth`, I am not sure why that happens (but I'm confident that doing the build and bundling with `Release` instead would work)

**Comment by nicogodet** [comment]
I'm not confident it would work
Could you double check your commands and see if your are consistent with keyword value for CMAKE_BUILD_TYPE and --config ?

**Comment by danodom-tcb** [comment]
@nicogodet @m-kuhn , everything has worked so far up till Step 4 above in this thread. 

1. this was the command from step3 above
cmake --build build --config RelWithDebInfo --parallel


2. this was the last command that failed (straight from  step 4 above)
 cmake --build build --config RelWithDebInfo --target bundle

**Comment by danodom-tcb** [comment]
@nicogodet @m-kuhn , It is a path issue... check this out..

this is the first error running the last step 
>>  CMake Error at C:/QGIS/QGIS/build/src/core/cmake_install.cmake:1046 (file):
    file INSTALL cannot find
    "C:/QGIS/QGIS/build/output/bin/Release/qgis_core.dll": File exists.
  Call Stack (most recent call first):
    C:/QGIS/QGIS/build/src/cmake_install.cmake:37 (include)
    C:/QGIS/QGIS/build/cmake_install.cmake:285 (include)

Check the actual path 👍 
![image](https://github.com/user-attachments/assets/45ff13b6-2fe7-48bf-b6e5-de2e7ca3d03e)


CMake is referencing it in "Release" but the file is located in "RelWithDebInfo"

**Comment by danodom-tcb** [comment]
@nicogodet is there a good way to pass in the actual path here, or is something hardcoded?

I know it's a hack, but I could copy the RelWithDebInfo -> Release folders as I think that might work.. Albeit not the preferred method.

**Comment by danodom-tcb** [comment]
> This is something configured by CMake. I ran into the same issue before because I mixed CMAKE_BUILD_TYPE and --config keyword value. I had to delete build folder and rebuild See my previous message

@nicogodet not completely following.. Are you saying I need to go back and rerun the first couple of steps.  This is the initial command that I ran.  

Not sure what you are requesting I do, but am hoping it's something with the last step where I don't have to do a complete rebuild. 

> 	cmake -S . 
		  -B build 
		  -D WITH_VCPKG=ON 
		  -D BUILD_WITH_QT6=ON 
		  -D WITH_QTWEBKIT=OFF 
		  -D VCPKG_TARGET_TRIPLET=x64-windows-release 
		  -D VCPKG_HOST_TRIPLET=x64-windows-release 
		  **-D CMAKE_BUILD_TYPE=RelWithDebInfo** 
		  -D WITH_DESKTOP=ON 
		  -D WITH_3D=ON 
		  -D WITH_BINDINGS=ON 
		  -D ENABLE_TESTS=OFF 
		  -D USE_CCACHE=ON 
		  -D CMAKE_C_COMPILER_LAUNCHER="C:\QGIS\ccache\ccache.exe" 
		  -D CMAKE_CXX_COMPILER_LAUNCHER="C:\QGIS\ccache\ccache.exe" 
		  -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:\QGIS\vcpkg\src" 
		  -D FLEX_EXECUTABLE="C:\QGIS\winflex\win_flex.exe" 
		  -D BISON_EXECUTABLE="C:\QGIS\winflex\win_bison.exe" 
		  -D CREATE_ZIP=ON

**Comment by nicogodet** [comment]
Well I can't help much...

If CMake target looks for files in Release folder it means that you run with Release at some point.

Indeed completely delete build folder and rebuild is a solution.

**Comment by danodom-tcb** [comment]
> Well I can't help much...
> 
> If CMake target looks for files in Release folder it means that you run with Release at some point.
> 
> Indeed completely delete build folder and rebuild is a solution.

I didn't run with Release, I have posted my exact commands from the shell and are verbatim to what you posted ^^^^^^.

**Comment by danodom-tcb** [comment]
> > Well I can't help much...
> > If CMake target looks for files in Release folder it means that you run with Release at some point.
> > Indeed completely delete build folder and rebuild is a solution.
> 


@nicogodet 
I didn't run with Release, I have posted my exact commands from the shell and are verbatim to what you posted ^^^^^^.

**Comment by danodom-tcb** [comment]
@nicogodet @m-kuhn , so for grins and giggles, I did the copy/paste for the RelWithDebInfo folder -> release, and step 3 ran...

AND... I am able to run QGIS using the last command..  my guess is that there's something in that initial cmake that is hardcoded to release in some cases, but not all.   IDK.. 

I am happy to help debug/triage this as it feels like the process refinements you guys have made are VERY, VERY close to beginning to end.

**Comment by danodom-tcb** [comment]
@m-kuhn @nicogodet , just bumping this in case you have any ideas or if this needs to be reported as a separate issue.

**Comment by danodom-tcb** [comment]
> I solved this issue by cleaning my build folder and retry. So it looks like a local issue rather than a real one...

@nicogodet , did you change anything on the command that you ran (see above)?  I ran from a clean system that had just been built out, so there wasn't any residual data/items floating around.

**Comment by m-kuhn** [comment]
I think this is about running from the build folder vs running from a bundled zip.
To run from the build folder, a qgis.env might help point it to the proper paths in the vcpkg_installed folder, but didn't check yet what variables to set exactly.

**Comment by danodom-tcb** [comment]
> I think this is about running from the build folder vs running from a bundled zip. To run from the build folder, a qgis.env might help point it to the proper paths in the vcpkg_installed folder, but didn't check yet what variables to set exactly.

@m-kuhn @nicogodet I am thinking that we can close this if you guys concur.  The only issue is the last step in building the artifacts and an issue of where it's trying to find them. 

Leave this open to track, or close? 

Either way, a huge thank you to you both for getting this almost to the finish line.

**Comment by m-kuhn** [comment]
Here is an approach that works for this scenario: https://github.com/kadas-albireo/kadas-albireo2/pull/179
Inspired by the current (windows specific) .env file of QGIS, adjusted for vcpkg

**Comment by xuhancn** [comment]
Mark, I occurred this issue also.

**Comment by NathanW2** [comment]
Following `qgis.env.in` file replaced with the values for  `qgis.env` makes it work for me

`qgis.env.in`
```
PATH=@VCPKG_BASE_DIR@\bin;C:\Windows\system32;C:\Windows;C:\Windows\system32\WBem
PYTHONPATH=%BUILD_OUTPUT_FOLDER@\python;@VCPKG_BASE_DIR@/tools/python3/Lib;@VCPKG_BASE_DIR@/tools/python3/Lib/site-packages
QGIS_PREFIX_PATH=@BUILD_OUTPUT_FOLDER@
PROJ_DATA=@VCPKG_BASE_DIR@/share/proj
QT_PLUGIN_PATH=@VCPKG_BASE_DIR@/Qt6/plugins
PYTHONHOME=@VCPKG_BASE_DIR@\tools\python3
```

Create `qgis.env` next to `qgis.exe` and replace `BUILD_OUTPUT_FOLDER` with your result build folder e.g `{your folder}\build\output` and `@VCPKG_BASE_DIR@` with the full path to the VCPKG installed folder

Haven't done much cmake in ages but will see if I can add it as part of the build step for SDK_PATH enabled to generate the qgis.env file.

**Comment by m-kuhn** [comment]
`VCPKG_APPLOCAL_DEPS` is nice in theory but in practice lacks some needed bits for the complex qgis ecosystem (e.g. data from gdal and proj or plugins for some dependencies)
For now we have (not super smart but working) logic to do that in https://github.com/qgis/QGIS/blob/master/cmake/VcpkgInstallDeps.cmake
